pkg/controller/deployment/util/deployment_util.go

```diff
@@ -416,6 +416,9 @@ func MaxUnavailable(deployment extensions.Deployment) int32 {
 	}
 	// Error caught by validation
 	_, maxUnavailable, _ := ResolveFenceposts(deployment.Spec.Strategy.RollingUpdate.MaxSurge, deployment.Spec.Strategy.RollingUpdate.MaxUnavailable, *(deployment.Spec.Replicas))
+	if maxUnavailable > *(deployment.Spec.Replicas) {
+		return *(deployment.Spec.Replicas)
+	}
```

Comment:
OK, sounds reasonable. Then should we put this logic in `MaxUnavailableInternal` too?